config/core/configmaps/gc.yaml

```
@@ -48,9 +48,17 @@ data:
     stale-revision-timeout: "15h"
 
     # Minimum number of generations of revisions to keep before considering
-    # them for GC
+    # them for GC.
     stale-revision-minimum-generations: "20"
 
+    # Maximum number of generations of revisions to keep before considering
+    # them for GC.  If creation or staleness time-bounds are set, the system
+    # will ignore those times to achieve this cap. -1 to disable a maximum bound.
```

```suggestion
    # Maximum number of generations of revisions to keep regardless of
    # creation or staleness time-bounds
    # Set to -1 to disable this setting.
```
Simpler? :)
Also how about using `0` as sentinel, since it's invalid setting anyway?

---

I should make it clearer that actively-referenced are not included in that count.

Actually I also don't like the name of this setting (Generation has a meaning in k8s that is a little different than this. I suppose they mean generations of the Configuration that created it?) but I suppose its too late to rename the existing stuff.